avocado/utils/nvme.py

```diff
@@ -90,3 +93,220 @@ def get_controller_id(controll_name):
+    namespaces = []
+    output = process.run(cmd, shell=True, sudo=True, ignore_status=True).stdout_text
+    for line in output.splitlines():
+        namespaces.append(int(line.split()[1].split("]")[0]) + 1)
+    return namespaces
+
+
+def get_current_ns_list(controller_name):
+    """
+    Returns the list of namespaces in the nvme controller
+
+    :param controller_name: Name of the nvme controller like nvme0, nvme1
+    :rtyp: list
+    """
+    namespace_list = []
+    namespaces_ids = get_current_ns_ids(controller_name)
+    for ns_id in namespaces_ids:
+        namespace_list.append(f"/dev/{controller_name}n{ns_id}")
+    return namespace_list
+
+
+def get_block_size(namespace):
+    """
+    Returns the block size of the namespace.
+    If not found, return defaults to 4k.
+
+    :param namespace: Name of the namespace like /dev/nvme0n1 etc..
+    :rtype: Integer
+    """
+    if namespace:
+        cmd = f"nvme id-ns /dev/{namespace}"
+        out = process.system(cmd, shell=True, ignore_status=True)
+        for line in out.splitlines:
+            if "in use" in line:
+                return pow(2, int(line.split()[4].split(":")[-1]))
+    return 4096
+
+
+def delete_ns(controller_name, ns_id):
+    """
+    Deletes the specified namespace on the controller
+
+    :param controller_name: Nvme controller name to which namespace belongs
+    :param ns_id: namespace id to be deleted
+    """
+    cmd = f"nvme delete-ns {controller_name} -n {ns_id}"
+    try:
+        process.system(cmd, shell=True, ignore_status=True)
+        LOGGER.debug("ns deleted successfully")
+    except process.CmdError as detail:
+        LOGGER.debug("failed to delete ns, more details= %s", detail)
+
+
+def delete_all_ns(controller_name):
+    """
+    Deletes all the name spaces available on the given nvme controller
+
+    :param controller_name: Nvme controller name eg : nvme0, nvme1 etc..
+    """
+    namespaces_ids = get_current_ns_ids(controller_name)
+    for ns_id in namespaces_ids:
+        delete_ns(controller_name, ns_id)
+
+
+def is_ns_exists(controller_name, ns_id):
+    """
+    Returns if that particular namespace exists on the controller or not
+
+    :param controller_name: name of the controller on which we want to check
+                            ns existance
+
+    :returns: True if ests else False
+    :rtype: boolean
+    """
+    ns_list = get_current_ns_list(controller_name)
+    if ns_id in ns_list:
+        return True
+    return False
+
+
+def get_lba(namespace):
+    """
+    Returns LBA of the namespace. If not found, return defaults to 0.
+
+    :param namespace: nvme namespace like /dev/nvme0n1, /dev/nvme0n2 etc..
+    :rtype: Integer
+    """
+    if namespace:
+        cmd = f"nvme id-ns /dev/{namespace}"
+        out = process.run(cmd, shell=True, ignore_status=True)
+        for line in out.splitlines():
+            if "in use" in line:
+                return int(line.split()[1])
+    return "0"
```

Comment:
you are returning one integer and if not found.. returning string "0"  .. make sure you return None

Reply:
or just return 0